kube/src/config/file_loader.rs

```diff
@@ -91,7 +86,9 @@ impl ConfigLoader {
         for named_user in config.auth_infos {
             if &named_user.name == user_name {
                 let mut user = named_user.auth_info.clone();
-                user.load_gcp().await?;
+                if let Some(provider) = &user.auth_provider {
+                    user.token = auth::token_from_provider(provider).await?;
+                }
                 user_opt = Some(user);
             }
         }
```

Comment:
Is this intentional (last one wins)?

Just finding the last one should be better then:
```rust
if let Some(named) = config.auth_infos.iter().rev().find(|a| &a.name == user_name) {
    let mut user = named.auth_info.clone();
    if let Some(provider) = &user.auth_provider {
        user.token = auth::token_from_provider(provider).await?;
    }
    user_opt = Some(user);
}
```

Reply:
I think it probably hasn't come up that there's multiple auth_infos with a matching user_name.
Might as well do the `find` and take the first match. Last seems counter-intuitive.